modules/auxiliary/scanner/scada/profinet_siemens.rb

```diff
@@ -0,0 +1,156 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'packetfu'
+
+class MetasploitModule < Msf::Auxiliary
+  def initialize
+    super(
+      'Name'        => 'Siemens Profinet Scanner',
+      'Description' => %q{
+          This module will use Layer2 packets, known as Profinet Discovery packets,
+        to detect all Siemens (and sometimes other) devices on a network.
+        It is perfectly SCADA-safe, as there will only be ONE single packet sent out.
+        Devices will respond with their IP configuration and hostnames.
+        Created by XiaK Industrial Security Research Center (www[dot]xiak[dot]be))
+      },
+      'References'  =>
+        [
+          [ 'URL', 'https://wiki.wireshark.org/PROFINET/DCP' ],
+          [ 'URL', 'https://github.com/tijldeneut/ICSSecurityScripts' ]
+        ],
+      'Author'      => 'Tijl Deneut <tijl.deneut[at]howest.be>',
+      'License'     => MSF_LICENSE
+      )
+
+    register_options(
+      [
+        OptString.new('INTERFACE', [ true, 'Set an interface', 'eth0' ]),
+        OptInt.new('TIMEOUT', [ true, 'Seconds to wait, set longer on slower networks', 2 ])
+      ], self.class
+    )
+  end
+
```

Comment:
I'm not sure, I found it very difficult selecting a method for raw packet handling supported in Ruby/Metasploit. Any ideas are welcome, thanks.